### PR TITLE
Create npm-publish.yml

### PR DESCRIPTION
**Pull request by tjvr**
This is copied from [GitHub's documenation](https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry).

I've already created the repository secret, and added a granular API key to it.

Have I missed any steps, or is it really this easy?

### CODE REVIEW

**Comment by apple502j** [comment]
Sorry for late review @tjvr! I think for npm publish this is okay. that said, I think we should automate all parts of the process - updating scratchblocks.github.io, releasing on GitHub, etc.

Also, fun fact: Time flies (and Node 16 is already at the end of LTS).

**Comment by tjvr** [comment]
I certainly agree we should automate updating scratchblocks.github.io!

How would you automate releasing on GitHub?

**Comment by apple502j** [comment]
@tjvr An action manually triggered from the Actions tab, which builds the files and publishes them on Releases along with auto-gened changelogs?

I think manual publishing is fine, but it is a possibility.

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    types: [published]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
```
Comment (apple502j) [inline .github/workflows/npm-publish.yml:13]:
```suggestion
          node-version: '18.x'
```

**Comment by tjvr** [comment]
I guess I will have to wait until we have another release to do before we can test this!

**Pull request merged**